resources/js/beatmapsets-show/scoreboard/controller.ts

```diff
@@ -65,13 +65,17 @@ export default class Controller {
       return 'unranked';
     }
 
-    if (!core.currentUser?.is_supporter && (this.currentType !== 'global' || this.enabledMods.size > 0)) {
+    if (!core.currentUser?.is_supporter && this.requiresSupporter) {
       return 'supporter_only';
     }
 
     return this.xhrState;
   }
 
+  get requiresSupporter() {
+    return !['global', 'team'].includes(this.currentType) || this.enabledMods.size > 0;
```

Comment:
can be a `const Set<ScoreboardType>` (also maybe make it match against supporter required instead of not free)